projects/ffmpeg/build.sh

```diff
@@ -0,0 +1,275 @@
+      --enable-static --disable-examples $THEORA_BUILD_ARGS
+make clean
+make -j$(nproc)
+make install
+
+cd /vorbis
+./autogen.sh
+./configure --prefix="$FFMPEG_DEPS_PATH" --enable-static
+make clean
+make -j$(nproc)
+make install
+
+cd /libxml2
+./autogen.sh --prefix="$FFMPEG_DEPS_PATH" --enable-static \
+      --without-debug --without-ftp --without-http \
+      --without-legacy --without-python
+make clean
+make -j$(nproc)
+make install
+
+# Remove shared libraries to avoid accidental linking against them.
+rm $FFMPEG_DEPS_PATH/lib/*.so
+rm $FFMPEG_DEPS_PATH/lib/*.so.*
+
+# Build ffmpeg.
+cd /ffmpeg
+FFMPEG_BUILD_ARGS=''
+
+PKG_CONFIG_PATH="$FFMPEG_DEPS_PATH/lib/pkgconfig" ./configure \
+        --cc=$CC --cxx=$CXX --ld="$CXX $LDFLAGS -std=c++11" \
+        --extra-cflags="-I$FFMPEG_DEPS_PATH/include" \
+        --extra-ldflags="-L$FFMPEG_DEPS_PATH/lib" \
+        --prefix="$FFMPEG_DEPS_PATH" \
+        --pkg-config-flags="--static" \
+        --libfuzzer=$LIB_FUZZING_ENGINE \
+        --optflags=-O1 \
+        --enable-gpl \
+        --enable-nonfree \
+        --enable-libass \
+        --enable-libfdk-aac \
+        --enable-libfreetype \
+        --enable-libopus \
+        --enable-libtheora \
+        --enable-libvorbis \
+        --enable-libvpx \
+        --enable-libxml2 \
+        --enable-nonfree \
+        --disable-muxers \
+        --disable-protocols \
+        --disable-demuxer=rtp,rtsp,sdp \
+        --disable-devices \
+        --disable-shared \
+        $FFMPEG_BUILD_ARGS
+make clean
+make -j$(nproc) install
+
+# Build target.
+if [[ $CONFIG = "sydr" || $CONFIG = "coverage" ]]
+then
+      $CC $LDFLAGS -c -o main.o /opt/StandaloneFuzzTargetMain.c
+      $CC $LDFLAGS -I/ffmpeg -c -o target_bsf_fuzzer.o /target_bsf_fuzzer.c
+      $CC $LDFLAGS -DFFMPEG_CODEC=AV_CODEC_ID_MPEG1VIDEO -DFUZZ_FFMPEG_VIDEO \
+                   -I/ffmpeg -c -o target_dec_fuzzer.o /target_dec_fuzzer.c
+      $CC $LDFLAGS -DIO_FLAT=0 -I/ffmpeg -c -o target_dem_fuzzer.o /target_dem_fuzzer.c
+fi
+
+# BSF target.
+$CC $LINK_FLAGS $TARGET_BSF
+
+# DEC target.
+$CC $LINK_FLAGS $TARGET_DEC
+
+# DEM target.
+$CC $LINK_FLAGS $TARGET_DEM
+
+# Get corpus.
+if [[ $CONFIG = "coverage" ]]
+then
+      # Download test samples, will be used as seed corpus.
+      # DISABLED.
+      # TODO: implement a better way to maintain a minimized seed corpora
+      # for all targets. As of 2017-05-04 now the combined size of corpora
+      # is too big for ClusterFuzz (over 10Gb compressed data).
+      export TEST_SAMPLES_PATH=/ffmpeg/fate-suite/
+      make fate-rsync SAMPLES=$TEST_SAMPLES_PATH
+      #python group_seed_corpus.py $TEST_SAMPLES_PATH /
+      rm -rf /corpus
+      mkdir /corpus
+      cd /ffmpeg/fate-suite
+      find * -iname *.m* >> fnd
+      find * -iname *.a* >> fnd
+      find * -iname *.h* >> fnd
+      find * -iname *.b* >> fnd
+      find * -iname *.f* >> fnd
+      while read p; do cp -r $p /corpus/; done < fnd
```

Comment:
It seems we should collect separate corporas for the targets. Maybe we can make initial python script work?

Reply:
this script works unclear. we decided to leave it for now. now in corpus directory there are all files collected from the initial corpus.